src/NetscriptWorker.ts

```diff
@@ -525,29 +527,30 @@ function createAndAddWorkerScript(
 
   // Create the WorkerScript. NOTE: WorkerScript ctor will set the underlying
   // RunningScript's PID as well
-  const s = new WorkerScript(runningScriptObj, pid, NetscriptFunctions);
-  s.ramUsage = oneRamUsage;
+  const workerScript = new WorkerScript(runningScriptObj, pid, NetscriptFunctions);
+  workerScript.ramUsage = oneRamUsage;
 
   // Add the WorkerScript to the global pool
-  workerScripts.set(pid, s);
+  workerScripts.set(pid, workerScript);
   WorkerScriptStartStopEventEmitter.emit();
 
   // Start the script's execution
-  let p: Promise<WorkerScript> | null = null; // Script's resulting promise
-  if (s.name.endsWith(".js") || s.name.endsWith(".ns")) {
-    p = startNetscript2Script(player, s);
+  let scriptExecution: Promise<WorkerScript> | null = null; // Script's resulting promise
+  if (workerScript.name.endsWith(".js") || workerScript.name.endsWith(".ns")) {
+    scriptExecution = startNetscript2Script(player, workerScript);
   } else {
-    p = startNetscript1Script(s);
-    if (!(p instanceof Promise)) {
+    scriptExecution = startNetscript1Script(workerScript);
+    if (!(scriptExecution instanceof Promise)) {
       return false;
     }
   }
 
   // Once the code finishes (either resolved or rejected, doesnt matter), set its
   // running status to false
-  p.then(function (w: WorkerScript) {
-    w.running = false;
-    w.env.stopFlag = true;
+  scriptExecution.then(function (w: WorkerScript) {
+    if(w !== workerScript) console.error("!BUG! Wrong WorkerScript instance !BUG!")
```

Comment:
How about 
```suggestion
    if(w !== workerScript) console.error("!BUG! Wrong WorkerScript instance !BUG! (This means we missed a reference somewhere)")
```

just to help the next person

Reply:
Cleaned the code up a bit by switching the `startNetscript*Script` functions to return `Promise<void>` instead of `Promise<WorkerScript>`.